openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -43,6 +43,25 @@
 invalid_not_found_fmt = "XBlock {usage_key} does not exist, or you don't have permission to view it."
 
 
+def parse_version_request(version_str: str | None) -> LatestVersion | int:
+    """
+    Given a version parameter from a query string (?version=14, ?version=draft,
+    ?version=published), get the LatestVersion parameter to use with the API.
```

Comment:
Does it make sense to make this an optional part of the URL pattern matching instead of a querystring? So something like:`xblocks/v2/<str:usage_key_str>;<str:version>/`. I mostly ask because it seems like this convention could eventually make its way into other calls that could have their own querystring handling, e.g. XBlock handler calls.

Reply:
@ormsbee Please let me know if that makes sense ^ and if you wanna take another look at this PR before I merge.

Reply:
@bradenmacdonald: Okay, I think that's fine. And if it becomes a problem, we can address it then. No need to re-review on my part. Thank you!

Reply:
@ormsbee You were right 😅

https://github.com/openedx/edx-platform/issues/35673